src/api/inline-api.ts

```diff
@@ -178,23 +178,25 @@ export class DataviewInlineApi {
      * Takes a filename and arbitrary input data.
      */
     public view(viewName: string, input: any) {
-        /** This cannot be used on systems without file access (i.e. web and mobile devices). */
-        if (!(this.app.vault.adapter instanceof FileSystemAdapter)) {
-            renderErrorPre( this.container, `Dataview: file system access is not available.` );
+
+        let currentFile = this.app.workspace.getActiveFile();
+        let sourcePath  = currentFile?.path ?? '';
+        
+        let viewPath = `${viewName}/view.js`;
+        let viewFile = this.app.metadataCache.getFirstLinkpathDest( viewPath, sourcePath );
+
+        /** Check that a file exists for the requested view name. */
+        if ( !viewFile ) {
+            renderErrorPre( this.container, `Dataview: view file not found.` );
             return;
         }
 
-        /** Check that a file exists for the requested view name. */
-        let viewPath = `${this.app.vault.configDir}/dataviews/${viewName}/view.js`;
-        this.app.vault.adapter.exists(viewPath).then(viewExists => {
-            if ( !viewExists ) throw new Error(`view file does not exist: ${viewPath}`);
-
-            return this.app.vault.adapter.read(viewPath);
-        }).then(viewData => {
-            /**
-             * Create a function from file contents. This is the dangerous part:
-             * it’s basically eval(). Consider adding sanitization & filtering.
-             */
+        /**
+         * Create a function from file contents. This is the dangerous part:
+         * it’s basically eval(). Consider adding sanitization & filtering.
+         */
+        this.app.vault.read( viewFile ).then(viewData => {
```

Comment:
You may want to add a `catch` to this for two reasons: (1) avoiding check-then-use race condition if someone deletes the view file right after you check if it resolves; and (2) in case the view function throws an error, so we can render an error message for users.